daemon/api_validate_test.go

```diff
@@ -0,0 +1,438 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/daemon"
+	"github.com/snapcore/snapd/overlord/assertstate"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+var _ = check.Suite(&apiValidationSetsSuite{})
+
+type apiValidationSetsSuite struct {
+	apiBaseSuite
+}
+
+func (s *apiValidationSetsSuite) SetUpTest(c *check.C) {
+	s.apiBaseSuite.SetUpTest(c)
+	d := s.daemon(c)
+	d.Overlord().Loop()
+	defer d.Overlord().Stop()
```

Comment:
Should this be `s.AddCleanup(d.Overlord().Stop)` ? Using a defer here that stops overlord as soon as the function exits? Or am I missing something? If so, maybe a short comment?